src/article-crawler.js

```diff
@@ -21,54 +21,72 @@ module.exports = async function (sourceConfigs, { maxArticlesPerPage, articleUrl
-				for (const articleUrl of articleUrls.slice(0, maxArticlesPerPage)) {
-					try {
-						await browserPage.goto(articleUrl, {
-							waitUntil: 'load',
-							// Remove the timeout
-							timeout: 0,
-						})
+				if(linkSelector){
+					const articleUrls = await browserPage.$$eval(linkSelector, (elements) => elements.map((element) => element.href))
+					for (const articleUrl of articleUrls.slice(0, maxArticlesPerPage)) {
+						try {
+							await browserPage.goto(articleUrl, {
+								waitUntil: 'load',
+								// Remove the timeout
+								timeout: 0,
+							})
 
-						const title = await browserPage.$eval(articleSelectors.title, (element) => element.textContent)
-						const excerpt = await browserPage.$eval(articleSelectors.excerpt, (element) => element.textContent)
-						const leadImage = await browserPage.$eval(articleSelectors['lead-image'], (element) => element.src)
-						let content = []
-						for (const contentSelector of articleSelectors.content) {
-							const innerContent = await browserPage.$$eval(contentSelector, (elements) => elements.map((element) => element.innerHTML))
-							content.push(innerContent)
-						}
+							const title = await browserPage.$eval(articleSelectors.title, (element) => element.textContent)
+							const excerpt = await browserPage.$eval(articleSelectors.excerpt, (element) => element.textContent)
+							const leadImage = await browserPage.$eval(articleSelectors['lead-image'], (element) => element.src)
+							let content = []
+							for (const contentSelector of articleSelectors.content) {
+								const innerContent = await browserPage.$$eval(contentSelector, (elements) => elements.map((element) => element.innerHTML))
+								content.push(innerContent)
+							}
 
-						content = htmlToText
-							.fromString(content, {
-								wordwrap: false,
-								ignoreImage: true,
-								ignoreHref: true,
-								ignoreImage: true,
-								preserveNewlines: false,
-							})
-							.trim()
-							.slice(0, 2000)
+							content = htmlToText
+								.fromString(content, {
+									wordwrap: false,
+									ignoreImage: true,
+									ignoreHref: true,
+									ignoreImage: true,
+									preserveNewlines: false,
+								})
+								.trim()
+								.slice(0, 2000)
+
+							let article = {
+								sourceName: source.sourceName,
+								category: page.category,
+								url: articleUrl,
+								title,
+								shortDescription: excerpt,
+								imageLink: leadImage,
+								isHeadline: true,
+								content: content,
+								createdDate: source.crawlTime,
+								modifiedDate: source.crawlTime,
+								publishedDate: source.crawlTime,
+								link: articleUrl,
+								topic: page.category,
+							}
 
+							articles.push(article)
+						} catch (error) {
+							console.log('error produced article crawl', error.message, articleUrl)
+						}
+					}
+				}else{
+					const imageSelector = page['image-selector']
```

Comment:
can you probably refactor this into a function? Also, can you also write some tests for it?